code/modules/antagonists/heretic/structures/knock_final.dm

```diff
@@ -0,0 +1,73 @@
+/obj/structure/knock_tear
```

Comment:
You'll probably want this to have infinite pull / push / move resist, also override singularity act, etc. Though I don't think it'll matter.